src/fdc/DiskManipulator.cc

```diff
@@ -346,11 +348,19 @@ void DiskManipulator::create(std::span<const TclObject> tokens)
 {
 	static_vector<unsigned, MAX_PARTITIONS> sizes;
 	size_t totalSectors = 0;
-	bool dos1 = false;
+	MSXBootSectorType bootType = MSXBootSectorType::DOS2;
 
 	for (const auto& token : view::drop(tokens, 3)) {
 		if (token == "-dos1") {
-			dos1 = true;
+			bootType = MSXBootSectorType::DOS1;
+			continue;
+		}
+		if (token == "-dos2") {
+			bootType = MSXBootSectorType::DOS2;
+			continue;
+		}
+		if (token == "-extended") {
+			bootType = MSXBootSectorType::EXTENDED;
 			continue;
 		}
```

Comment:
Hmm, here we again convert a string into a `MSXBootSectoType` enum value.

Maybe we should factor out the helper I used above (move the `parseType` lambda function out and make it a proper free function). Then use it here like:
```
if (auto t = parseBootSectorType(token)) {
    bootType = *t;
    continue;
}
```